framework/docker/internal/busybox.go

```diff
@@ -86,20 +45,20 @@ func EnsureBusybox(ctx context.Context, cli *client.Client) error {
 		return nil
 	}
 
-	// Use retry mechanism for pulling the busybox image
-	err = retryWithBackoff(ctx, func() error {
+	// Use wait.ForCondition to retry pulling the busybox image
+	err = wait.ForCondition(ctx, 60*time.Second, 5*time.Second, func() (bool, error) {
 		rc, err := cli.ImagePull(ctx, BusyboxRef, dockerimagetypes.PullOptions{})
 		if err != nil {
-			return fmt.Errorf("pulling busybox image: %w", err)
+			return false, fmt.Errorf("pulling busybox image: %w", err)
```

Comment:
I think we'll need to do

```go
return false, nil
```

here, if an error is returned from the polling condition it is propagated immediately.

Reply:
yea, I pushed a fix